src/lib/api.ts

```diff
@@ -26,11 +33,20 @@ const login = async (username: string, password: string) => {
 
   const response = await fetch(`${API_URL}/user/login`, {
     method: "POST",
-    body: JSON.stringify({ username, password, token }),
-    headers,
+    body: JSON.stringify({ username, password }),
+    credentials: "include",
+    headers: {
+      "Content-Type": "application/json",
+      "X-CSRF-Token": token,
+    },
   });
 
-  return { isLogged: response.status < 400, token };
+  const data = (await response.json()) as LoginResponse;
+
+  return {
+    isLogged: response.status < 400,
+    token: data.token,
```

Comment:
Gdyby data przyszło puste, to może się wywalić, więc proponuję tak się zabezpieczyć:

```suggestion
    token: data?.token,
```